CONTRIBUTING.txt

```diff
@@ -0,0 +1,98 @@
+    6. To provide an encouraging environment where contributors learn and improve their skills
+    7. To protect the free and open nature of the FreeCAD project
+
+2. Fundamentals
+
+    1. FreeCAD uses the git distributed revision control system.
+    2. Source code for the main application and related subprojects is hosted on github.com in the FreeCAD organization, hereafter referred to as 'the platform'.
+    3. Problems are discrete, well-defined limitations or bugs.
+    4. FreeCAD uses the platform's issue-tracking system to track problems
+    5. Patches are sets of code changes that resolve a single problem
+    6. FreeCAD uses the Pull Request workflow for evaluating and accepting patches
+
+
+3. Roles
+    1. "User": An member of the wider FreeCAD community who uses the software.
+    2. "Contributor":  A person who submits a patch that resolves a previously identified problem. Contributors do not have commit access to the repository unless they are also maintainers. Everyone, without distinction or discrimination, SHALL have an equal right to become a Contributor
+    3. "Maintainer": A person who merges patches. Maintainers may or may not be Contributors. Their role is to enforce the process. Maintainers have commit access to the repository.
+    4. "Administrator": Administrators have additional authority to maintain the list of designated maintainers.
+
+4. Licensing, Ownership, and Credit
+    1. FreeCAD uses the Lesser General Public License, version 2, or superior (LGPL2+) for all source code.  Additional details can be found in the LICENSE file.
+    2. All contributions to FreeCAD SHALL use the same license as the project.
+    3. All contributions are owned by their authors unless assigned to another.
+    4. FreeCAD does not have a mandatory copyright assignment policy.
+    5. A Contributor who wishes to be identified in the Credits section of the application "About" dialog is responsible for identifying themselves appropriately by notifying an administrator or maintainer.
+
+5. Contribution Requirements
+
+    1. Patches are submitted in the form of Pull Requests (PR)
+    2. Maintainers and Contributors MUST have a platform account and SHOULD use their real names or a well-known alias. If the platform alias differs from the account alias on the FreeCAD Forum, effort SHOULD be taken to avoid confusion.
+    3. A PR SHOULD be a minimal and accurate answer to exactly one identified and agreed-on problem.
+    4. A PR SHOULD refrain from adding additional dependencies to the FreeCAD project unless no other option is available.
+    5. Code submissions MUST adhere to the code style guidelines of the project if these are defined.
+    6. If a PR contains multiple commits, each commit MUST compile cleanly and add value to the history of the project. Checkpoint commits should be squashed.
+    7. A PR SHALL NOT include non-trivial code from other projects unless the Contributor is the original author of that code.
```

Comment:
This is much like adding a new dependency in terms of the possible effects on licensing issues.  At minimum the code from another project needs to be reviewed for suitability and compatible licensing.  This is not a trivial decision by a single Contributor but a FreeCAD-wide issue that has to be carefully considered.

Reply:
Are you suggesting a change to the text?

Reply:
Would it make sense to add commentary along the line of "reviewers should consider these ideas when reviewing this type of change"?  Otherwise, no change.

Reply:
I'm going to leave it as is.  Reviewers should take everything into account.  (And I'm SURE they will :-P  )